src/overlays/actors/ovl_En_Kusa/z_en_kusa.c

```diff
@@ -7,72 +16,485 @@
+            EffectSsGRipple_Spawn(globalCtx, &contactPos, 400, 800, 4);
+            EffectSsGRipple_Spawn(globalCtx, &contactPos, 500, 1100, 8);
+            this->actor.minVelocityY = -3.0f;
+            *D_80A9C1D4 = (*D_80A9C1D4 >> 1);
+            *D_80A9C1D0 = (*D_80A9C1D0 >> 1);
+            *D_80A9C1DC = (*D_80A9C1DC >> 1);
+            *D_80A9C1D8 = (*D_80A9C1D8 >> 1);
+            this->actor.bgCheckFlags = this->actor.bgCheckFlags & 0xFFBF;
+            Audio_PlaySoundAtPosition(globalCtx, &this->actor.posRot, 40, NA_SE_EV_DIVE_INTO_WATER_L);
+        }
+        func_80A9B140(&this->actor);
+        Math_ApproxS(D_80A9C1D4, *D_80A9C1D0, 0x1F4);
+        Math_ApproxS(D_80A9C1DC, *D_80A9C1D8, 0xAA);
+        this->actor.shape.rot.x += *D_80A9C1D4;
+        this->actor.shape.rot.y += *D_80A9C1DC;
+        func_80A9B174(&this->actor.velocity, 0.05f);
+        func_8002D7EC(&this->actor);
+        func_8002E4B4(globalCtx, this, 7.5f, 35.0f, 0.0f, 0xC5);
+        Collider_CylinderUpdate(&this->actor, &this->collider);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9BA98.s")
+void func_80A9BEAC(EnKusa* this) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9BAD8.s")
+    if ((this->actor.params & 3) != 1) {
+        if ((this->actor.params & 3) == 2) {
+            // This function is blank
+            func_80A9AFA0(this, func_80A9BF30);
+        }
+    } else {
+        func_80A9AFA0(this, func_80A9BEFC);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9BBB0.s")
+void func_80A9BEFC(EnKusa* this, GlobalContext* globalCtx) {
+    if (this->unk_19C >= 0x78) {
+        func_80A9C00C(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9BC1C.s")
+void func_80A9BF30(EnKusa* this, EnKusaActionFunc actionFunc) {
+    // Function was intentionally left blank?...
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9BEAC.s")
+void func_80A9BF3C(EnKusa* this) {
+    this->actor.posRot.pos.x = this->actor.initPosRot.pos.x;
+    this->actor.posRot.pos.y = this->actor.initPosRot.pos.y - 9.0f;
+    this->actor.posRot.pos.z = this->actor.initPosRot.pos.z;
+    // Set scale factors
+    func_80A9B1FC(this);
+    this->actor.shape.rot = this->actor.initPosRot.rot;
+    func_80A9AFA0(this, func_80A9BFA8);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9BEFC.s")
+void func_80A9BFA8(EnKusa* this, GlobalContext* globalCtx) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9BF30.s")
+    if (this->unk_19C >= 0x79) {
+        if ((Math_ApproxF(&this->actor.posRot.pos.y, this->actor.initPosRot.pos.y, 0.6f) != 0) &&
+            (this->unk_19C >= 0xAA)) {
+            func_80A9C00C(this);
+        }
+    }
```

Comment:
```suggestion
       if ((this->unk_19C > 200) && Math_ApproxF(&this->actor.posRot.pos.y, this->actor.initPosRot.pos.y, 0.6f) && (this->unk_19C >=170)) {
        func_80A9C00C(this);
    }
```
These conditionals can combine into one, and MathApproxF is a boolean.